proto/deadlock.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+package deadlockpb;
+
+import "gogoproto/gogo.proto";
+
+message WaitForEntriesRequest {
+}
+
+message WaitForEntriesResponse {
+	repeated WaitForEntry entries = 1 [(gogoproto.nullable) = false];
+}
+
+message WaitForEntry {
+	// The transaction id that is waiting.
+	uint64 txn = 1;
+	// The transaction id that is being waited for.
+	uint64 wait_for_txn = 2;
+	// The hash value of the key is being waited for.
+	uint64 key_hash = 3;
```

Comment:
If there is hash conlfict, does it matter?

Reply:
Then it will timeout, it's very rare, so I think it's acceptable.

Reply:
No, in this case, it doesn't matter, because there is definitely a deadlock.

Reply:
The key_hash can be used to make deadlock only retry a single statement.